lib/Twig/Extension/Core.php

```diff
@@ -1139,7 +1139,15 @@ function twig_convert_encoding($string, $to, $from)
  */
 function twig_length_filter(Twig_Environment $env, $thing)
 {
-    return is_scalar($thing) ? mb_strlen($thing, $env->getCharset()) : count($thing);
+    if (is_scalar($thing)) {
+        return mb_strlen($thing, $env->getCharset());
+    }
+    
+    if (is_object($thing) && !($thing instanceof \Countable) && is_callable([$thing, '__toString'])) {
```

Comment:
short array syntax is a no-go in the Twig codebase. We stay consistent with the 1.x branch as it is maintained.